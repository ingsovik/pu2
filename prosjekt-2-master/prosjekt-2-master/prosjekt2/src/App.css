/**************/
/*  Coloring  */
/**************/

body{
    background-color: white;
    color: black;
    overflow-x:hidden;
}

#root, .App, .inputDiv{
    background-color: inherit;
    color: inherit;
}
button{
    border: 1px solid rgb(113, 154, 190);
    background-color: rgb(221, 238, 253);
    color: black;
}

button:hover{
    border: 1px solid black;
}

.issueInfoDiv, .commitInfoDiv{
    border: 1px solid rgb(113, 154, 190);
    background-color: rgb(221, 238, 253);
    color: black;
}


#tglColorBtn{
    margin-left: 10px;
    padding: 15px 50px;
    font-weight: bold;
    text-decoration: underline;
}
#tglColorBtn:hover{
    background-color: white;
}


/*******************/
/*  Inputs/submit  */
/*******************/

.inputDiv {
    width: 100%;
    position: fixed;
    top: 0;
    z-index: 2;
    display: flex;
    justify-content: center;
    padding: 20px 0 20px 0;
    border-bottom: 1px solid grey;
}

.inputs {
    width: 240px;
}

#tokenLabel, #urlLabel {
    padding-right: 5px;
}

.labelDiv {
    display: flex;
    align-items: center;
}

#urlInput {
    margin-right: 20px;
}

#submitBtn {
    text-align: center;
    margin-left: 15px;
    padding: 6px 10px;
    width: 70px;
    border-radius: 3px;
}

#submitBtn:hover {
    background-color: white;
    color: black;
    border: 1px solid black;
}

@media screen and (max-width: 950px) {
    .inputDiv {
        display: flex;
        flex-direction: column;
        align-items: center;
    }
    .inputs {
        margin-top: 10px;
    }
    #urlInput {
        margin-right: 0;
    }
    #tokenLabel, #urlLabel {
        padding: 0;
    }
    #tokenLabel {
        margin-top: 20px;
    }
    #submitBtn {
        margin-left: 0;
    }


    .projectInfoDiv {
        padding-top: 230px !important;
    }
    #infoHs{
        font-size: 80%;
    }
}


/*****************/
/*  ProjectInfo  */
/*****************/

p {
    line-height: 100%;
}

.projectInfoDiv {
    display: flex;
    justify-content: center;
    padding-top: 104px;
}
#infoHs{
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin-left: 10px;
}
#infoName{
    margin-top: 10px;
}
#svgPic{
    max-width: 30vw;
    max-height: 30vw;
    min-width: 100px;
    padding: 0;
}



/**********************/
/*  ToggleBtn + Grid  */
/**********************/

.toggleBtn {
    padding: 15px 30px;
    font-weight: bold;
    width: 200px;
    text-decoration: underline;
}

.tglBtnDiv {
    display: flex;
    justify-content: center;
    margin-top: 3%;
}

.tglH1 {
    text-align: center;
    margin: 50px 0 50px 0;
}

/* .toggleDiv { 
    /*Everything: ProjectInfo, togglebtns, graphs, grid
} */

/* .dateContext { /*Datepicker + h1 + graph
} */

#datePicker { /*Datepicker*/
    text-align: center;
}

#dateContextProviderGraph { /*Graph + h1*/
    display: flex;
    justify-content: center;
}

#pickDateLabel {
    margin-right: 5px;
    font-weight: 500;
    font-size: 120%;
}

#issueTglButtons {
    display: flex;
    justify-content: center;
    margin-bottom: 30px;
}

.issuesGrid, .commitsGrid {
    display: flex;
    justify-content: center;
    margin: 30px 0;
}

.gridDiv {
    width: 80%;
    display: grid;
    grid-template-columns: repeat(3, minmax(0, 1fr));
    grid-gap: 8px;
}

@media screen and (max-width: 900px) {
    .gridDiv {
        width: 80%;
        display: flex;
        flex-direction: column;
    }
}

.issueInfoDiv, .commitInfoDiv {
    padding: 20px;
    min-width: 33%;
}

.issueInfoDiv:hover, .commitInfoDiv:hover, .toggleBtn:hover {
    background-color: white;
    color: black;
}

.issueSpan, .commitSpan, .infoText {
    margin-left: 3px; /* Space between "Title:" and "Title of issue" */
}