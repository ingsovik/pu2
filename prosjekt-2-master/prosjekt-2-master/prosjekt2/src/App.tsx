import React from "react";
import "./App.css";
import { InputProvider } from './context/inputContext';
import { ToggleView } from './components/toggleView';


function App() {
  
  return (   
    <>               
      <div className="App">
        <InputProvider>
          <ToggleView />
        </InputProvider>
      </div>
    </>
    
  );
}

export default App;

