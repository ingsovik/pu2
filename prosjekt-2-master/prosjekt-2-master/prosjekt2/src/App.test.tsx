import * as React from 'react';
import App from './App';
import Issues from './components/issues';
import renderer from 'react-test-renderer'
import Commits from './components/commits';


// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

test('Issue renders', () =>{
    const tree = renderer.create(<Issues />)
    expect(tree).toMatchSnapshot()
})

describe('Test asynch fetch works', () =>{
    const mockURL = 'https://gitlab.stud.idi.ntnu.no/api/v4/projects/11877/issues'
    const mockData = [{ id: '12341', iid: '1', title: 'Test issue'
        ,state: 'open',
        created_at: {date: '2021-09-16'},
        closed_at: {date: '2021-09-29'}}];
        const getData = jest.fn(url => mockData)
        it('returns data for api call',()=> {
            expect(getData(mockURL)).toBe(mockData)
        });
        it('called getData with mockURL', () =>{
            expect(getData).toHaveBeenCalledWith(mockURL)
        })
})