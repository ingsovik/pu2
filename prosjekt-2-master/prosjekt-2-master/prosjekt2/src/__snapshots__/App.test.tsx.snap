// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue renders 1`] = `
<div
  className="issuesGrid"
>
  <div
    className="gridDiv"
  >
    <div
      className="loadText"
    >
      Loading...
    </div>
  </div>
</div>
`;
